Archeo: add CSS for mobile menu font size

mikachan opened the pull request:
<!-- Thanks for contributing to our free themes! Please provide as much information as possible with your Pull Request by filling out the following - this helps make reviewing much quicker! -->

#### Changes proposed in this Pull Request:
This increases the font size of the main navigation on small screens, as suggested in https://github.com/Automattic/themes/issues/5631:

Before|After
----|-----
![image](https://user-images.githubusercontent.com/1645628/157496818-cda6dbc5-4f9e-4333-9964-5c355b948d71.png)| ![image](https://user-images.githubusercontent.com/1645628/157496690-da74e451-3f07-46f9-ac0f-9ba92cf3d183.png)

The `blockGap` has already been reduced to `var(--wp--custom--spacing--small)` in https://github.com/Automattic/themes/pull/5640. This is what I've used in the above screenshot. Should this be reduced any further?

#### Related issue(s):
https://github.com/Automattic/themes/issues/5631

mikachan committed:
Add CSS for mobile menu font size

jffng commented on archeo/style.css:
```diff
@@ -224,3 +224,13 @@ body > .is-root-container > .wp-block-template-part > .wp-block-cover,
+ * Possible future fix: https://github.com/WordPress/gutenberg/pull/33543
+ */
+@media (max-width: 599px) {
+	.wp-block-navigation__responsive-container-content li {
```
I noticed this selector doesn't work in the editor for non-standard links. For example, I have Submenu and a Home link, and the large font size isn't applying: 
<img width="1904" alt="Screen Shot 2022-03-09 at 2 12 32 PM" src="https://user-images.githubusercontent.com/5375500/157514605-3f786a8d-a89c-4d38-a850-f5bf0b2d394a.png">

It makes me wonder if adding this CSS is worth it, or if it's going to cause more confusion if a user changes this navigation's font size in the editor and previews it. 

Another thought — could we create a custom CSS variable that uses a responsive type size and apply it to the navigation block [here](https://github.com/Automattic/themes/blob/trunk/archeo/theme.json#L140). That way any user modifications to the font-size would still take precedence.

pbking replied:
> could we create a custom CSS variable that uses a responsive type size and apply it to the navigation block [here](https://github.com/Automattic/themes/blob/trunk/archeo/theme.json#L140).

Oh that's a fine idea.

mikachan replied:
I think the CSS variable with a responsive type size is a great idea, however, I'm not sure how to do this where the font size is bigger on smaller screens and smaller on larger screens.

For now, I've gone with setting the larger font size for the overlay in CSS (we were already doing this, I missed it when I was working on this PR!) and setting the default font size in theme.json.

kjellr commented:
If this doesn't already exist, we should file an issue to ask for the ability to specify an overlay-specific font size. I imagine this will be a common use case, and we already have the ability to set overlay-specific values for color via `overlayBackgroundColor` and `"overlayTextColor`.

mikachan committed:
Remove CSS patch

mikachan committed:
Move navigation font size into theme.json

pbking commented:
I don't see the responsive menu "small" as posted in the before shots, I looks correct as of [this change](https://github.com/Automattic/themes/pull/5462).  The only change I can see from this PR is that the font size configuration has been moved from the BLOCK to THEME.JSON which looks good to me.

mikachan commented:
> The only change I can see from this PR is that the font size configuration has been moved from the BLOCK to THEME.JSON

Yeah this sounds correct, I missed that we'd already set the large font size for the responsive menu via CSS, but then it looks like it was being overwritten by the font size set in the header markup. So, I've removed the font size from the `header.html` and moved it to theme.json.

Going to look for / create an issue for setting an `overlayFontSize` (or similar) setting in GB now..

pbking commented:
I couldn't find a related issue so have opened this: https://github.com/WordPress/gutenberg/issues/39352

The change looks good, 🚢

jffng commented:
> Going to look for / create an issue for setting an overlayFontSize (or similar) setting in GB now..

Looks like some folks are starting to think about this holistically here: https://github.com/WordPress/gutenberg/issues/39142
